src/main/java/com/shapesecurity/shift/es2017/codegen/CodeGen.java

```diff
@@ -226,6 +227,12 @@ public CodeRep reduceAssignmentTargetWithDefault(@Nonnull AssignmentTargetWithDe
         return seqVA(binding, factory.token("="), init);
     }
 
+    @Nonnull
+    @Override
+    public CodeRep reduceAwaitExpression(@Nonnull AwaitExpression node, @Nonnull CodeRep expression) {
+        return seqVA(factory.token("await"), expression);
+    }
+
```

Comment:
Also need to codegen the `async` for async function expressions / declarations / methods.